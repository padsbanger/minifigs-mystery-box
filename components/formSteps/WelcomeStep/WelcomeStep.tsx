import Button from "@/components/ui/Button/Button";
import Heading from "@/components/ui/Heading/Heading";
import Image from "next/image";

import {
  useIsLoadingMinifigs,
  useMinifigsStore,
} from "@/lib/utils/services/MinifigsService/useMinifigsStore/useMinifigsStore";

const WelcomeStep = () => {
  const getMinifigs = useMinifigsStore((state) => state.getMinifigs);
  const setCurrentStep = useMinifigsStore.getState().setCurrentStep;
  const isLoading = useIsLoadingMinifigs();

  const handleStart = async () => {
    await getMinifigs();
    setCurrentStep("CHOOSE");
  };

  return (
    <div className="flex flex-col items-center h-screen justify-center">
      <div className="relative">
        <Image
          src={"/question-mark.svg"}
          alt="question-mark"
          className="absolute md:-top-20 md:-left-20 -rotate-[5deg] -top-20 left-20"
          width={64}
          height={64}
        />
        <Image
          src={"/question-mark.svg"}
          alt="question-mark"
          className="absolute md:-bottom-20 md:-right-20 rotate-[25deg] -bottom-10 right-10"
          width={48}
          height={48}
        />
        <Heading
          text="lego minifigs mystery box"
          variant={"heading1"}
          level={1}
          classes="my-4 text-center text-white"
        />
        <Button
          label={"Let`s go!"}
          disabled={isLoading}
          isLoading={isLoading}
          onClick={handleStart}
          classes="flex m-auto"
        />
      </div>
    </div>
  );
};

export default WelcomeStep;
