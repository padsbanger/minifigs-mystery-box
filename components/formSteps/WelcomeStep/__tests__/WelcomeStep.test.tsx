import { render } from "@testing-library/react";
import WelcomeStep from "../WelcomeStep";

const setup = () => render(<WelcomeStep />);

describe("WelcomeStep View", () => {
  test("should match snapshot", () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
