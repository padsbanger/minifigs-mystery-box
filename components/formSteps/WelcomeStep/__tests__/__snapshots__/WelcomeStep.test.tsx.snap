// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeStep View should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center h-screen justify-center"
  >
    <div
      class="relative"
    >
      <img
        alt="question-mark"
        class="absolute md:-top-20 md:-left-20 -rotate-[5deg] -top-20 left-20"
        data-nimg="1"
        decoding="async"
        height="64"
        loading="lazy"
        src="/question-mark.svg"
        style="color: transparent;"
        width="64"
      />
      <img
        alt="question-mark"
        class="absolute md:-bottom-20 md:-right-20 rotate-[25deg] -bottom-10 right-10"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="/question-mark.svg"
        style="color: transparent;"
        width="48"
      />
      <h1
        class="font-plus-jakarta-sans leading-normal uppercase text-headline-1 font-700 leading-[120%] my-4 text-center text-white"
      >
        lego minifigs mystery box
      </h1>
      <button
        class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-6 px-12 rounded-md disabled:opacity-55 shadow-sm hover:shadow-xl transition-all active:shadow-2xl flex m-auto"
      >
        <span
          class="uppercase font-plus-jakarta-sans"
        >
          Let\`s go!
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
